x-pack/plugins/cloud_security_posture/public/pages/findings/findings.tsx

```diff
@@ -4,17 +4,42 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
 import React from 'react';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import type { EuiPageHeaderProps } from '@elastic/eui';
+import { EuiLoadingSpinner } from '@elastic/eui';
 import { FindingsTableContainer } from './findings_container';
 import { CspPageTemplate } from '../../components/page_template';
+import { useKubebeatDataView } from './utils';
+import { TEST_SUBJECTS } from './constants';
+
+const pageHeader: EuiPageHeaderProps = {
+  pageTitle: 'Findings',
+};
+
+export const Findings = () => {
+  const dataView = useKubebeatDataView();
+
+  if (dataView.status === 'loading') return <LoadingPrompt />;
+
+  return (
+    <CspPageTemplate pageHeader={pageHeader}>
+      {dataView.status === 'success' && dataView.data && (
+        <FindingsTableContainer dataView={dataView.data} />
+      )}
+      {(dataView.status === 'error' || !dataView.data) && <ErrorPrompt />}
+    </CspPageTemplate>
+  );
+};
```

Comment:
by the looks of it, when entering this page the entire screen would rerender (blink), including the side bar (due to `LoadingPrompt` being outside of the `CspPageTemplate`.
I'm guessing that is by choice and not a mistake but I think having the page templates renders while displaying the loading prompt inside it is a better ux choice. will result in less jumpiness when going from loading state to success/error state and when navigating between CSP pages.

```suggestion
export const Findings = () => {
  const dataView = useKubebeatDataView();

// callback or out of component scope
const render = () => {
  if (dataView.status === 'loading') return <LoadingPrompt />;
  if (dataView.status === 'error' || !dataView.data) return <ErrorPrompt />;
  if (dataView.status === 'success' && dataView.data) return <FindingsTableContainer dataView={dataView.data} />;
}

  return (
    <CspPageTemplate pageHeader={pageHeader}>
      {render()}
    </CspPageTemplate>
  );
};
```

Reply:
changed this to include the EmptyPrompt inside the template. 

didn't use a render function.

although i find them more readable, i'm still not 100% sure there isn't a difference from a rendering perspective, as inlining the components and calling a function that returns them (even memoized) has (i think) slightly different JSX semantics. 


also, this is probably going to change  the agent would or BE would register the `kubebeat` index and dataview,

Reply:
> && dataView.data' can be removed now since it was checked at the line above

🚀